### PR TITLE
Update custom backend docs

### PR DESCRIPTION
**Pull request by williamwen42**
Title.

### CODE REVIEW

File: docs/source/dynamo/custom-backends.rst [hunk 1]
```diff
@@ -1,8 +1,95 @@
...
+to define custom backends.
+
+A backend function has the contract
+``(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]) -> Callable``.
+
+Backend functions can be called by TorchDynamo, the graph tracing component of ``torch.compile``,
+after tracing an FX graph and are
+expected to return a compiled function that is equivalent to the traced FX graph.
+The returned callable should have the same contract as the ``forward`` function of an ``nn.Module``:
+``(*args: Any) -> Any``.
```
Comment (jansel) [inline docs/source/dynamo/custom-backends.rst:17]: This isn't exaclty right.  The contract will should match `gm: torch.fx.GraphModule` and will go from all tensor inputs to a list of tensors as output.

File: docs/source/dynamo/custom-backends.rst [hunk 2]
```diff
@@ -1,8 +1,95 @@
...
+
+* You can pass a string containing your backend function's name to ``torch.compile`` instead of the function itself,
+  for example, ``torch.compile(model, backend="my_compiler")``.
+* It is required for use with the `minifier <https://pytorch.org/docs/master/dynamo/troubleshooting.html>`__. Any generated
+  code from the minifier must call your code that registers your backend function, typically through an ``import`` statement.
+
+Custom Backends after AOTAutograd
+---------------------------------
+
+If you want your backend to be called by AOTAutograd rather than TorchDynamo, you can wrap your backend with
```
Comment (jansel) [inline docs/source/dynamo/custom-backends.rst:63]: Add a sentence about why someoneone would want this (support training, normalized IRs)

**Review by jansel — CHANGES_REQUESTED** [review]

**Review by jansel — APPROVED** [review]
